model/model_training/check_dataset_appearances.py

```diff
@@ -5,23 +5,73 @@
+
+python check_dataset_appearances.py -d alpaca_gpt4 vicuna gpteacher_roleplay red_pajama wizardlm_70k --cache_dir .cache --mode sft
+python check_dataset_appearances.py -d wizardlm_70k --cache_dir .cache --mode sft
+
+python check_dataset_appearances.py -d alpaca_gpt4 vicuna gpteacher_roleplay wizardlm_70k joke poem_instructions oa_stackexchange tell_a_joke --cache_dir .cache --mode sft
+python check_dataset_appearances.py joke  --cache_dir .cache --mode sft
+
+python check_dataset_appearances.py -d webgpt gpt4all code_alpaca minimath humaneval_mbpp_codegen_qa humaneval_mbpp_testgen_qa grade_school_math_instructions recipes cmu_wiki_qa oa_wiki_qa_bart_10000row prosocial_dialogue explain_prosocial soda oa_leet10k dolly15k --cache_dir .cache --mode sft
+python check_dataset_appearances.py -d soda oa_leet10k dolly15k --cache_dir .cache --mode sft
 """
 import argparse
 import pprint
 from collections import defaultdict
 
+from model_training.check_dataset_counts import Mode
 from model_training.custom_datasets import get_one_dataset
-from model_training.custom_datasets.entities import Mode
-from model_training.custom_datasets.formatting import DatasetEntry
-from model_training.custom_datasets.qa_datasets import (
-    re_reference_remove,
-    re_single_reference_remove,
-    re_whitespace_newline_match,
-)
-from model_training.custom_datasets.utils import FILTER_BY_WORDS
-
-RE_TO_CHECK = [re_whitespace_newline_match, re_reference_remove, re_single_reference_remove]
-STRINGS_TO_CHECK = [*FILTER_BY_WORDS]
+from model_training.custom_datasets.formatting import DatasetEntryLm, DatasetEntrySft
+
+# from model_training.custom_datasets.qa_datasets import (
+#     re_reference_remove,
+#     re_single_reference_remove,
+#     re_whitespace_newline_match,
+# )
+# from model_training.custom_datasets.utils import FILTER_BY_WORDS
+
+RE_TO_CHECK = []  # [re_whitespace_newline_match, re_reference_remove, re_single_reference_remove]
+# STRINGS_TO_CHECK = [*FILTER_BY_WORDS]
+STRINGS_TO_CHECK = [
+    "as a language model",
+    "as an AI language model",
+    "As a large language model",
+    "As an AI ",
+    "an AI language model you don't have",
+    "As an AI language model, I cannot",
+    "As an AI language model, I do not",
+    "As an AI language model, I am not able",
+    "As an AI language model, I don't have personal",
+    "I am an AI language model and do not",
+    "As an AI language model, I don't have",
+    "As an AI language model, I am only able",
+    "AI language model and I do not",
+    "As an AI language model, I cannot modify",
+    "As an AI language model, I do not",
+    "I know as an AI language model you don't have",
+    "as an AI language model, you cannot",
+    "I'm sorry, but as an AI language model",
+    "As an AI language model, I don't have",
+    "I'm an AI ",
+    "I am an AI ",
+    "my name is OpenAI",
+    "trained by OpenAI",
+    "as an OpenAI language model",
```

Comment:
+1, maybe we could even consider removing any reference to OpenAI altogether

Reply:
I'll remove `OpenAI` altogether. Found some example like `Best regards, your OpenAI language model` and other things. It's probably safer to remove it altogether